src/Sarif.Driver/Sdk/MultithreadedAnalyzeCommandBase.cs

```diff
@@ -485,6 +485,10 @@ private static void LogCachingLogger(TContext globalContext, CachingLogger cachi
 
             if (results?.Count > 0)
             {
+                globalContext.Logger.FileRegionsCache = cachingLogger.FileRegionsCache ?? new FileRegionsCache();
```

Comment:
@michaelcfanning Why are we duplicating this thrice inside the next three if-blocks. Could we not do this once immediately after IEnumerated artifact = globalContext.currentTarget?